java/memory/memory-core/src/main/java/org/apache/arrow/memory/Accountant.java

```diff
@@ -73,12 +74,15 @@ public Accountant(Accountant parent, String name, long reservation, long maxAllo
     this.allocationLimit.set(maxAllocation);
 
     if (reservation != 0) {
+      if (parent == null) {
+        throw new IllegalArgumentException(String.valueOf("Accountant(parent) must not be null"));
+      }
```

Comment:
Why do you use String.valueOf?

```suggestion
      Preconditions.checkArgument(parent != null, "parent must not be null");
```

Reply:
As I mention at https://github.com/apache/arrow/pull/37723#discussion_r1444982298 (1st option) if I added Preconditions.xxx then that finished also with errors, then, is needed to add inline throw validation.

Finished OK:
````
final BufferLedger oldLedger = map.remove(allocator);
if (oldLedger == null) {
  throw new IllegalArgumentException(String.valueOf("BufferLedger(oldLedger) must not be null"));
}
BufferAllocator oldAllocator = oldLedger.getAllocator();
````

Finished with errors:
````
final BufferLedger oldLedger = map.remove(allocator);
Preconditions.checkArgument(oldLedger != null, "BufferLedger(oldLedger) must not be null");
BufferAllocator oldAllocator = oldLedger.getAllocator();

Error message:
error: [dereference.of.nullable] dereference of possibly-null reference oldLedger
````

Reply:
You need to annotate the Preconditions methods so that Checker Framework recognizes them...

Reply:
Changed